### PR TITLE
Add pty/pipe hangup test

### PR DESCRIPTION
**Pull request by lgrahl**
Test for https://github.com/creytiv/re/pull/159

Testing on other platforms than Linux would be appreciated. :slightly_smiling_face:

### CODE REVIEW

File: src/pty.c [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	test->err = err;
+	re_cancel();
+}
+
+
+static void pty_hdlr(int flags, void *arg)
+{
+	struct pty_test *test = arg;
+	(void) flags;
+	abort_test(test, 0);
```
Comment (alfredh) [inline src/pty.c]: perhaps the function should check `flags` and abort the test if `FD_EXCEPT` flag is set ?

when running the test, I am getting many `FD_READ` events, and in some cases an `FD_EXCEPT`
at the end.

Comment (lgrahl) [reply src/pty.c]: I'm not expecting any input but I agree that this would probably be cleaner.

**Comment by alfredh** [comment]
testing on OSX:

```
$ ./retest -m select -r test_pty
single testcase: test_pty
retest: libre version 0.6.0 (x86_64/darwin)
using async polling method 'select'
regular tests:       test_pty: flags: 1
test_pty: read size: 0
OK

$ ./retest -m kqueue -r test_pty
single testcase: test_pty
retest: libre version 0.6.0 (x86_64/darwin)
using async polling method 'kqueue'
regular tests:       test_pty: flags: 1
test_pty: read size: 0
OK
```

File: src/pty.c [hunk 2]
```diff
@@ -0,0 +1,118 @@
+#define _BSD_SOURCE
+#define _DEFAULT_SOURCE
+#include <errno.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <signal.h>
+#include <pty.h>
```
Comment (alfredh) [inline src/pty.c:7]: on osx we need to include <util.h> instead

File: src/pty.c [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+{
+	test->err = err;
+	re_cancel();
+}
+
+
+static void pty_hdlr(int flags, void *arg)
+{
+	struct pty_test *test = arg;
+	int err;
```
Comment (alfredh) [inline src/pty.c:34]: nit; `int err` is uninitialized.

Comment (lgrahl) [reply src/pty.c:34]: I think it's always set before it's evaluated. Do you want me to initialise it with `0`?

File: src/pty.c (hunk 2 above)
Comment (lgrahl) [reply src/pty.c:7]: Instead of `pty.h`? So:

```c
#ifdef DARWIN
#include <util.h>
#else
#include <pty.h>
#endif
```

**Comment by alfredh** [comment]
thanks for the test. it has served its purpose, as I was able to test the re PR.

I dont think we need to add this code to master, as all the tests have to be fully
portable.

there is also no other tests that fork the process, so lets keep it like that for now.

thanks for your contributions anyway :)

**Pull request closed without merging**